nuxeo-runtime/nuxeo-runtime-tomcat-adapter/src/main/java/org/nuxeo/runtime/tomcat/NuxeoDeployer.java

```diff
@@ -89,47 +97,48 @@ protected void preprocess(Container container) {
             File deployerJar = FrameworkBootstrap.findFileStartingWidth(bundles, "nuxeo-runtime-deploy");
             File commonJar = FrameworkBootstrap.findFileStartingWidth(bundles, "nuxeo-common");
             if (deployerJar == null || commonJar == null) {
-                System.out.println("Deployer and/or common JAR (nuxeo-runtime-deploy* | nuxeo-common*) not found in "
-                        + bundles);
-                return;
-            }
-            ArrayList<URL> urls = new ArrayList<URL>();
-            File[] files = lib.listFiles();
-            if (files != null) {
-                for (File f : files) {
-                    if (f.getPath().endsWith(".jar")) {
-                        urls.add(f.toURI().toURL());
-                    }
+                if (log.isInfoEnabled()) {
+                    log.info("Deployer and/or common JAR (nuxeo-runtime-deploy* | nuxeo-common*) not found in "
+                            + bundles);
                 }
+                return;
             }
-            files = bundles.listFiles();
-            if (files != null) {
-                for (File f : files) {
-                    if (f.getPath().endsWith(".jar")) {
-                        urls.add(f.toURI().toURL());
-                    }
-                }
+            List<URL> urls = new ArrayList<>();
+            try (Stream<Path> p1 = Files.list(lib.toPath()); Stream<Path> p2 = Files.list(bundles.toPath())) {
+                Stream.concat(p1, p2).filter(p -> p.endsWith(".jar")).map(this::toURL).forEach(urls::add);
             }
             urls.add(homeDir.toURI().toURL());
             urls.add(new File(homeDir, "config").toURI().toURL());
-            try (URLClassLoader cl = new URLClassLoader(urls.toArray(new URL[urls.size()]), parentCl)) {
-                System.out.println("# Running Nuxeo Preprocessor ...");
+            try (URLClassLoader cl = new URLClassLoader(urls.toArray(new URL[0]), parentCl)) {
+                if (log.isInfoEnabled()) {
+                    log.info("# Running Nuxeo Preprocessor ...");
```

Comment:
Please don't change the `System.out.println` to `log.info` in this ticket. This warrants a separate ticket and examination of the needed output in this early preprocessing phase.

Reply:
I created a ticket for it https://jira.nuxeo.com/browse/NXP-24592